rotkehlchen/chain/ethereum/manager.py

```diff
@@ -306,26 +273,29 @@ def default_call_order(self, skip_etherscan: bool = False) -> List[NodeName]:
         """
         result = []
         if NodeName.OWN in self.web3_mapping:
-            result.append(NodeName.OWN)
+            # Assigning one is just a default since we always use it
+            # The weight is only important for the other nodes since they
+            # are selected suing this paramenter
+            result.append(WeightedNode(node=NodeName.OWN, weight=ONE))
 
-        selection = list(OPEN_NODES)
+        open_nodes = self.database.get_settings().nodes_to_connect
+        selection = list(open_nodes)
         if skip_etherscan:
-            selection.remove(NodeName.ETHERSCAN)
+            selection = [wnode for wnode in open_nodes if wnode.node != NodeName.ETHERSCAN]
```

Comment:
Why change this? I think that list.remove() is optimized by python itself compared to a list comprehension.

Or is it cause you now have the weight in there and can't identify it in any other way?

Reply:
yes, is because you have to check the property and is not direct as before. As far as I know this is the fastest solution in those cases